great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -1554,373 +1594,351 @@ def _get_interactive_expect_column_values_to_be_between_chart(
+            y_axis_title = "Column Values"
+        else:
+            y_axis_title = None
+
+        metric_plot_component: MetricPlotComponent
+        metric_plot_components: List[MetricPlotComponent] = []
+        for sanitized_metric_name in sanitized_metric_names:
+            metric_plot_component: MetricPlotComponent = MetricPlotComponent(
+                name=sanitized_metric_name,
+                alt_type=AltairDataTypes.QUANTITATIVE.value,
+                axis_title=y_axis_title,
+            )
+            metric_plot_components.append(metric_plot_component)
 
         domain_plot_component: DomainPlotComponent = DomainPlotComponent(
             name="column",
             alt_type=AltairDataTypes.NOMINAL.value,
         )
 
-        min_value_plot_component: ExpectationKwargPlotComponent = (
+        expectation_kwarg_plot_components: List[ExpectationKwargPlotComponent] = []
+
+        expectation_kwarg_plot_components.append(
             ExpectationKwargPlotComponent(
                 name=min_value,
                 alt_type=AltairDataTypes.QUANTITATIVE.value,
-                metric_plot_component=metric_plot_component,
+                axis_title=y_axis_title,
             )
         )
-        max_value_plot_component: ExpectationKwargPlotComponent = (
+        expectation_kwarg_plot_components.append(
             ExpectationKwargPlotComponent(
                 name=max_value,
                 alt_type=AltairDataTypes.QUANTITATIVE.value,
-                metric_plot_component=metric_plot_component,
-            )
-        )
-        strict_min_plot_component: ExpectationKwargPlotComponent = (
-            ExpectationKwargPlotComponent(
-                name=strict_min,
-                alt_type=AltairDataTypes.NOMINAL.value,
-                metric_plot_component=metric_plot_component,
-            )
-        )
-        strict_max_plot_component: ExpectationKwargPlotComponent = (
-            ExpectationKwargPlotComponent(
-                name=strict_max,
-                alt_type=AltairDataTypes.NOMINAL.value,
-                metric_plot_component=metric_plot_component,
+                axis_title=y_axis_title,
             )
         )
 
-        df: pd.DataFrame = pd.DataFrame(
-            columns=[
+        possible_df_columns: List[str] = (
+            [
                 batch_name,
             ]
             + batch_identifiers
+            + list(sanitized_metric_names)
             + [
-                metric_name,
                 column_name,
                 min_value,
                 max_value,
                 strict_min,
                 strict_max,
+                quantiles,
+                allow_relative_error,
             ]
         )
 
+        df_columns: List[str]
+        df: Optional[pd.DataFrame] = None
         for _, column_df in column_dfs:
-            df = pd.concat([df, column_df], axis=0)
+            df_columns = column_df.columns.intersection(possible_df_columns)
```

Comment:
My understanding is that this would be seen as redefining the variable for each loop (yes I understand the type is ignored). [PEP 526](https://peps.python.org/pep-0526/#global-and-local-variable-annotations) recommends to add the annotation before the loop.

Reply:
I don't think I understand. Aren't you redefining this variable each time? Also, I don't see that part of 526 which you are referring to. I see a section about doing this for the loop variables (`column_df` here).

Reply:
Also, this is just a nit and I'm fine either way.